lamindb/curators/__init__.py

```diff
@@ -545,6 +553,38 @@ def save_artifact(
         )
 
 
+def _assign_var_fields_categoricals_multimodal(
+    table_key: str | None,
+    table_slot: str,
```

Comment:
I think you are calling them based only spatialdata names, which is a bit confusing to call them table since table is often related to dataframe.

Also since they can be split from `slot` with `:`, why do you need these two params in addition to `slot`?

Reply:
> I think you are calling them based only spatialdata names, which is a bit confusing to call them table since table is often related to dataframe.

Renamed to modality and slot_type.

> Also since they can be split from slot with :, why do you need these two params in addition to slot?

We don't need them but it keeps the code shorter in the method. We have to get the modality and slot_type before this already anyways and should not repeat that in the assign function. I played a bit with it but I think the way I did it here is better despite us having to pass a few more parameters.